includes/admin/onboarding/onboarding.php

```diff
@@ -0,0 +1,637 @@
+require_once EDD_PLUGIN_DIR . 'includes/admin/onboarding/steps/step-configure_emails.php';
+require_once EDD_PLUGIN_DIR . 'includes/admin/onboarding/steps/step-tools.php';
+require_once EDD_PLUGIN_DIR . 'includes/admin/onboarding/steps/step-products.php';
+
+/**
+ * EDD_Onboarding Class.
+ *
+ * Takes care of everything related to Onboarding Wizard.
+ *
+ * @since 3.2
+ */
+class OnboardingWizard {
+
+	/**
+	 * Current Onboarding step.
+	 *
+	 * @since 3.2
+	 *
+	 * @var string
+	 */
+	private $current_step = 'business_info';
+
+	/**
+	 * Current Onboarding step index.
+	 *
+	 * @since 3.2
+	 *
+	 * @var int
+	 */
+	private $current_step_index = 1;
+
+	/**
+	 * Onboarding steps.
+	 *
+	 * @since 3.2
+	 *
+	 * @var array
+	 */
+	private $onboarding_steps = array();
+
+	/**
+	 * True if user started onboarding process.
+	 *
+	 * @since 3.2
+	 *
+	 * @var bool
+	 */
+	private $onboarding_started = false;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @since 3.2
+	 */
+	public function __construct() {
+		if ( ! is_admin() || wp_doing_cron() ) {
+			return;
+		}
+
+		// If Onboarding was already completed, abort.
+		if ( get_option( 'edd_onboarding_completed', '' ) !== '' ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'add_menu_item' ) );
+		add_action( 'admin_menu', array( $this, 'add_menu_item_class' ) );
+
+		// Abort if we are not requesting Onboarding Wizard.
+		if ( ! empty( $_REQUEST['page'] ) && 'edd-onboarding-wizard' !== wp_unslash( $_REQUEST['page'] ) ) {
+			return;
+		}
+
+		// Stripe calls are marked with onboardingWizard request parameter.
+		if ( empty( $_REQUEST['page'] ) && empty( $_REQUEST['onboardingWizard'] )  ){
+			return;
+		}
+
+		add_action( 'admin_init', array( $this, 'load_onboarding_wizard' ) );
+
+		// Ajax handlers.
+		add_action( 'wp_ajax_edd_onboarding_started', array( $this, 'ajax_onboarding_started' ) );
+		add_action( 'wp_ajax_edd_onboarding_load_step', array( $this, 'ajax_onboarding_load_step' ) );
+		add_action( 'wp_ajax_edd_onboarding_completed', array( $this, 'ajax_onboarding_completed' ) );
+		add_action( 'wp_ajax_edd_onboarding_skipped', array( $this, 'ajax_onboarding_skipped' ) );
+	}
+
+	/**
+	 * Add Onboarding Wizard submenu page.
+	 *
+	 * @since 3.2
+	 */
+	public function add_menu_item() {
+		add_submenu_page( 'edit.php?post_type=download', __( 'Setup', 'easy-digital-downloads' ), __( 'Setup', 'easy-digital-downloads' ), 'manage_shop_settings', 'edd-onboarding-wizard', array( $this, 'onboarding_wizard_sub_page' ) );
+	}
+
+	/**
+	 * Add class to the Onboarding Wizard subpage menu item.
+	 *
+	 * @since 3.2
+	 */
+	public function add_menu_item_class() {
+		global $submenu;
+
+		if ( empty( $submenu['edit.php?post_type=download'] ) ) {
+			return;
+		}
+
+		$edd_submenu     = $submenu['edit.php?post_type=download'];
+		$onboarding_menu = __( 'Setup', 'easy-digital-downloads' );
+
+		foreach ( $edd_submenu as $key => $value ) {
+			if ( $onboarding_menu == $value[0] ) {
```

Comment:
I'm not sure we can rely on this comparison since it's a translatable string (although presumably it would be translated both in the original and here for the comparison...but makes me a little nervous).